sql/analyzer/fix_exec_indexes.go

```diff
@@ -429,42 +428,50 @@ func (s *idxScope) visitSelf(n sql.Node) error {
 			s.expressions = append(s.expressions, fixExprToScope(e, scopes...))
 		}
 	case *plan.InsertInto:
-		rightSchema := make(sql.Schema, len(n.Destination.Schema())*2)
 		// schema = [oldrow][newrow]
-		for oldRowIdx, c := range n.Destination.Schema() {
-			rightSchema[oldRowIdx] = c
-			newRowIdx := len(n.Destination.Schema()) + oldRowIdx
-			if values, ok := n.Source.(*plan.Values); ok {
-				// The source table is either named via VALUES(...) AS ... or has
-				// the default value planbuilder.OnDupValuesPrefix
+		destSch := n.Destination.Schema()
```

Comment:
i've gone through line by line and i'm maybe missing if there was any logical change, correct me if i'm wrong